src/html2pdf.js

```diff
@@ -32,356 +32,412 @@
+                el.style.display = 'block';
+                var clientRect = el.getBoundingClientRect();
+                el.style.height = pxPageHeight - (clientRect.top % pxPageHeight) + 'px';
+            }, this);
+
+            // Return the container.
+            return container;
+        };
+
+        html2pdf.makePDF = function (canvas, pageSize, opt) {
+            // Calculate the number of pages.
+            var ctx = canvas.getContext('2d');
+            var pxFullHeight = canvas.height;
+            var pxPageHeight = Math.floor(canvas.width * pageSize.inner.ratio);
+            var nPages = Math.ceil(pxFullHeight / pxPageHeight);
+            var pageModulus = pxFullHeight % pxPageHeight;
+
+            // Create a one-page canvas to split up the full image.
+            var pageCanvas = document.createElement('canvas');
+            var pageCtx = pageCanvas.getContext('2d');
+            var pageHeight = pageSize.inner.height;
+            pageCanvas.width = canvas.width;
+            pageCanvas.height = pxPageHeight;
+
+            // Initialize the PDF.
+            var pdf = new jsPDF(opt.jsPDF);
+
+            /**
+             * added support for page split only on white "rows".
+             * check if pixel is not "white" if it is less than whiteColorLevel value for the rgb values
+             * if the row isn't "white" - will go one pixel up and test that row, up to max. number of maxPixelToGoUp rows up
+             *
+             */
+            var whiteColorLevel = 150;
+            var maxPixelToGoUp = 150;
+            var pageStartPos = 0;
+            for (var page = 0; page < nPages; page++) {
+                // Trim the final page to reduce file size.
+                if (page === nPages - 1) {
+                    pageCanvas.height = pageModulus;
+                    pageHeight = pageCanvas.height * pageSize.inner.width / pageCanvas.width;
+                }
+                // Display the page.
+                var w = pageCanvas.width;
+                var heightDrawBack = 0;
+                do {
+
+                    var h = pageCanvas.height - heightDrawBack;
+                    var whiteRow = true;
+                    var wPix = 0;
+                    var bPixCount = 0;
+
+                    while (wPix < w && whiteRow) {
+                        var pixelData = ctx.getImageData(wPix, pageStartPos + h, 1, 1).data;
+                        if (pixelData[0] < whiteColorLevel || pixelData[1] < whiteColorLevel || pixelData[2] < whiteColorLevel) {
+                            bPixCount++;
+                            if (bPixCount > 1) {
+                                whiteRow = false;
+                            }
+                        }
+                        wPix++;
+                    }
+                    heightDrawBack++;
+                }
+                while (!whiteRow && heightDrawBack < maxPixelToGoUp) ;
+                heightDrawBack--;
```

Comment:
At this point, if no white row was found, it looks like `heightDrawBack` will be at `maxPixelToGoUp`. Doesn't it need to be reset to 0 if `!whiteRow`?

Reply:
you mean, that if it didn't find white row up to *heightDrawBack*, it might as well cut the page on the first row?

Reply:
Yeah, if it still hasn't found a white row then it might as well use as much of the page as possible.